camel/functions/web_search.py

```diff
@@ -0,0 +1,243 @@
+    """
+    # https://developers.google.com/custom-search/v1/overview
+    GOOGLE_API_KEY = os.getenv("GOOGLE_API_KEY")
+    # https://cse.google.com/cse/all
+    SEARCH_ENGINE_ID = os.getenv("SEARCH_ENGINE_ID")
+
+    # using the first page
+    start = 1
+    # different language may get different result
+    language = "en"
+    # how many pages to return
+    numbers = 10
+    # constructing the URL
+    # doc: https://developers.google.com/custom-search/v1/using_rest
+    url = f"https://www.googleapis.com/customsearch/v1?" \
+          f"key={GOOGLE_API_KEY}&cx={SEARCH_ENGINE_ID}&q={query}&start=" \
+          f"{start}&lr={language}&num={numbers}"
+
+    responses = []
+    # make the get
+    try:
+        result = requests.get(url)
+        data = result.json()
+
+        # get the result items
+        search_items = data.get("items")
+
+        # iterate over 10 results found
+        for i, search_item in enumerate(search_items, start=1):
+            try:
+                long_description = \
+                    search_item["pagemap"]["metatags"][0]["og:description"]
+            except KeyError:
+                long_description = "N/A"
+            # get the page title
+            title = search_item.get("title")
+            # page snippet
+            snippet = search_item.get("snippet")
+
+            # extract the page url
+            link = search_item.get("link")
+            response = {
+                "Result_id": i,
+                "Title": title,
+                "Description": snippet,
+                "Long_description": long_description,
+                "URL": link
+            }
+            responses.append(response)
+
+    except requests.RequestException:
+        responses.append({"erro": "google search failed."})
+
+    return responses
+
+
+def text_extract_from_web(url: str) -> str:
+    r"""Get the text information from given url.
+
+    Args:
+        url (string): The web site you want to search.
+
+    Returns:
+        string: All texts extract from the web.
+    """
+    try:
+        # request the target page
+        response_text = requests.get(url).text
+
+        # parse the obtained page
+        soup = BeautifulSoup(response_text, features="html.parser")
+
+        for script in soup(["script", "style"]):
+            script.extract()
+
+        text = soup.get_text()
+        # strip text
+        lines = (line.strip() for line in text.splitlines())
+        chunks = (phrase.strip() for line in lines
+                  for phrase in line.split("  "))
+        text = ".".join(chunk for chunk in chunks if chunk)
+
+    except requests.RequestException:
+        text = f"can't access {url}"
+
+    return text
+
+
+# Split a text into smaller chunks of size n
+def create_chunks(text: str, n: int) -> List[str]:
+    r"""Returns successive n-sized chunks from provided text."
+
+    Args:
+        text: what need to be cut.
+        n: max length of chunk.
```

Comment:
Same, to add the variable type and make the explanation clearer. Remember to capitalize the first letter of the statement ("The max length of a single chunk.")